modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlAsynchronousApiTest.java

```diff
@@ -68,18 +65,13 @@ public void pageSequence() {
         var p0 = ars0.currentPage();
         AsyncResultSet<SqlRow> ars1 = await(ars0.fetchNextPage());
         var p1 = ars1.currentPage();
-        AsyncResultSet<SqlRow> ars2 = await(ars1.fetchNextPage().toCompletableFuture());
+        AsyncResultSet<SqlRow> ars2 = await(ars1.fetchNextPage());
         var p2 = ars2.currentPage();
         AsyncResultSet<SqlRow> ars3 = await(ars1.fetchNextPage());
         var p3 = ars3.currentPage();
         AsyncResultSet<SqlRow> ars4 = await(ars0.fetchNextPage());
         var p4 = ars4.currentPage();
 
-        assertSame(ars0, ars1);
-        assertSame(ars0, ars2);
-        assertSame(ars0, ars3);
-        assertSame(ars0, ars4);
```

Comment:
These checks no longer work because we now wrap the `AsyncResultSet` in an `AntiHijackAsyncResultSet` every time (see `PublicApiThreadingIgniteSql#doAsyncOperationForResultSet`)
```
org.opentest4j.AssertionFailedError: 
Expected :org.apache.ignite.internal.sql.api.AntiHijackAsyncResultSet@127302ec
Actual   :org.apache.ignite.internal.sql.api.AntiHijackAsyncResultSet@16b578a3
```